lib/pbench/server/api/resources/users_api.py

```diff
@@ -453,22 +456,32 @@ def put(self, target_username):
                 HTTPStatus.BAD_REQUEST,
                 message="Invalid fields in update request payload",
             )
-        protected = set(user_payload.keys()).intersection(set(User.get_protected()))
+        # Only admin user will be allowed to change other user's role. However,
+        # Admin users will not be able to change their admin role,
+        # This is done to prevent last admin user from de-admining him/herself
+        protected_db_fields = (
+            User.get_protected()
+            if self.auth.token_auth.current_user().is_admin()
+            and self.auth.token_auth.current_user() != result.target_user
+            else User.get_protected() + ["role"]
+        )
```

Comment:
While I do love the "ternary" operator, I found this code hard to check.  I would suggest something more like the following:
```
        protected_db_fields = User.get_protected()
        if (not self.auth.token_auth.current_user().is_admin() or
            self.auth.token_auth.current_user() == result.target_user):
            protected_db_fields.append("role"])
```
(But don't bother unless you have other changes to make.)